apps/vaporgui/SettingsParams.cpp

```diff
@@ -139,6 +139,15 @@ void SettingsParams::Reinit() { Init(); }
 
 SettingsParams::~SettingsParams() {}
 
+void SettingsParams::_swapTildeWithHome(std::string &file) const
+{
+    size_t pos = 0;
+    while ((pos = file.find("~", pos)) != std::string::npos) {
+        file.replace(pos, 1, QDir::homePath().toStdString());
+        pos += 1;
```

Comment:
This will work, but shouldn't 'pos' be incremented by QDir::homePath().toStdString().length()?  Not likely, but if QDir:: homePath() returns an empty string I'm not sure what would happen.

Reply:
Infinite loop problem understood.  Increment changed to 

`pos += QDir::homePath().toStdString().length();`